lib/ohai/plugins/windows/dmi.rb

```diff
@@ -0,0 +1,56 @@
+#
+# Author:: Pete Higgins (pete@peterhiggins.org)
+# Copyright:: Copyright (c) 2020 Kurt Yoder
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+=begin
+Here are the 5 sections of node['dmi'] that are returned on my Ubuntu workstation, and what I believe are the Win32 equivalents.
+
+chassis: https://docs.microsoft.com/en-us/windows/win32/cimwin32prov/win32-systemenclosure
+processor: https://docs.microsoft.com/en-us/windows/win32/cimwin32prov/win32-processor
+bios: https://docs.microsoft.com/en-us/windows/win32/cimwin32prov/win32-bios
+system: https://docs.microsoft.com/en-us/windows/win32/cimwin32prov/win32-computersystemproduct
+base_board: https://docs.microsoft.com/en-us/windows/win32/cimwin32prov/win32-baseboard
+
+I think these are the equivalents.
+
+Note the first one is what is in the system_enclosure plugin today. We can copy that pattern into the dmi plugin and use it to pull all of the sections. The system_enclosure plugin should then be set for future deprecation.
+=end
+
+Ohai.plugin(:DMI) do
+  provides "dmi"
+
+  DMI_TO_WIN32OLE = {
+    chassis: "SystemEnclosure",
```

Comment:
Since we already have the chassis plugin we probably want to just depend on that plugin and then use the data that we collect. 

Here's a depends line:

https://github.com/chef/ohai/blob/master/lib/ohai/plugins/linux/virtualization.rb#L21

Here's where we use the DMI data we've already collected there: https://github.com/chef/ohai/blob/master/lib/ohai/plugins/linux/virtualization.rb#L122

If you want to pull in the entire systemenclosure WMI class here then you should pull it here and then depend on this plugin in the system chassis plugin and use the data from this plugin there. Either way let's make sure we don't query the same data twice since WMI queries are slow.

Reply:
The implementation here was not meant to be final, more of an example to get feedback if I had the right idea of what the end result should be. More details in this issue: https://github.com/chef/ohai/issues/1359